src/DGtal/helpers/Shortcuts.h

```diff
@@ -1761,6 +1758,110 @@ namespace DGtal
+      /// positions as an OBJ file and a material MTL file. Optionnaly
+      /// you can specify the diffuse colors.
+      ///
+      /// @param[in] positions the bases of the vectors of the vector field.
+      /// @param[in] vf the vector field (an array of real vectors).
+      /// @param[in] thickness the thickness of the sticks representing the vector field.
+      /// @param[in] diffuse_colors either empty or a vector of size `normals.size()` specifying the diffuse color for each face.
+      /// @param[in] objfile the output filename.
+      /// @param[in] ambient_color the ambient color of all vectors.
+      /// @param[in] diffuse_color the diffuse color of all vectors (if diffuse_colors is empty).
+      /// @param[in] specular_color the specular color of all vectors.
+      /// @return 'true' if the output stream is good.
+      ///
+      /// @note It outputs only the vector field, not the surface onto which it is defined (if any).
+      static bool
+        saveOBJ
+	( const RealPoints&  positions,
+	  const RealVectors& vf,
+	  double             thickness,
+          const Colors&      diffuse_colors,
+          std::string        objfile,
+          const Color&       ambient_color  = Color( 32, 32, 32 ),
+          const Color&       diffuse_color  = Color( 200, 200, 255 ),
+          const Color&       specular_color = Color::White )
+        {
+          BOOST_STATIC_ASSERT (( KSpace::dimension == 3 ));
+          std::string mtlfile;
+          auto lastindex = objfile.find_last_of(".");
+          if ( lastindex == std::string::npos )
+            {
+              mtlfile  = objfile + ".mtl";
+              objfile  = objfile + ".obj";
+            }
+          else
+            {
+              mtlfile  = objfile.substr(0, lastindex) + ".mtl"; 
+            }
+          std::ofstream output_obj( objfile.c_str() );
+          output_obj << "#  OBJ format" << std::endl;
+          output_obj << "# DGtal::saveOBJ" << std::endl;
+          output_obj << "o vectors" << std::endl;
+          output_obj << "mtllib " << mtlfile << std::endl;
+          std::ofstream output_mtl( mtlfile.c_str() );
+          output_mtl << "#  MTL format"<< std::endl;
+          output_mtl << "# generated from MeshWriter from the DGTal library"<< std::endl;
+	  // Output vertices
+	  unsigned int n = std::min( positions.size(), vf.size() );
+	  for ( unsigned int i = 0; i < n; ++i )
+	    {
+	      RealPoint    p0 = positions[ i ];
+	      RealPoint    p1 = p0 + vf[ i ];
+	      unsigned int mc = std::max_element( vf.begin(), vf.end() ) - vf.begin();
+	      RealVector   b  =
+		mc == 2 ? RealVector( 1, 0, 0 ) :
+		mc == 1 ? RealVector( 0, 0, 1 ) : RealVector( 0, 1, 0 );
+	      RealVector   e0 = vf[ i ].crossProduct( b ).getNormalized();
+	      RealVector   e1 = vf[ i ].crossProduct( e0 ).getNormalized();
+	      RealPoint    t[4] = {  thickness * e0,  thickness * e1,
+				    -thickness * e0, -thickness * e1 };
```

Comment:
Well.. I don't understand.. You are not using the `vn` to export the normals ?

Reply:
This is a bit confusing w.r.t. to the doc and the usage, I think

Reply:
you're projecting the vertices onto the vf plane ?

Reply:
ok wait.. I'm confused by the overloading

Reply:
I would probably have renamed to some specific `saveVectorFieldOBJ`  (not great BTW) and you should probably describe what you mean by "saving a VF as OBJ". Let me check the VCM output

Reply:
ok.. small tubes..

Reply:
Done. I have changed the method name.